lib/kino/smart_cell/chart_builder.ex

```diff
@@ -74,6 +74,14 @@ defmodule Kino.SmartCell.ChartBuilder do
   @impl true
   def handle_info({:scan_binding_result, data_options, vl_alias}, ctx) do
     ctx = assign(ctx, data_options: data_options, vl_alias: vl_alias)
+
+    updated_fields =
+      case {ctx.assigns.fields["data_variable"], data_options} do
+        {nil, [%{variable: data_variable} | _]} -> updates_for_data_variable(ctx, data_variable)
+        _ -> %{}
+      end
+
+    ctx = update(ctx, :fields, &Map.merge(&1, updated_fields))
     broadcast_event(ctx, "set_available_data", %{"data_options" => data_options})
```

Comment:
We still need to send the fields and `setValues(fields)` on the client, right?

```suggestion
    broadcast_event(ctx, "set_available_data", %{"data_options" => data_options, "fields" => updated_fields})
```